__tests__/app.test.js

```diff
@@ -0,0 +1,47 @@
+const app = require('../app');
+const request = require('supertest');
+const db = require('../db/connection');
+const testData = require('../db/data/test-data');
+const seed = require('../db/seeds/seed');
+
+beforeEach(() => seed(testData));
+
+afterAll(() => db.end());
+
+// describe('error handling', ()=> {
+//     test('server respondds with 404 if given ?', () => {
+
+//     })
+// })
+
+describe('api', () => {
+    test('get 200 response from server', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then(response => {
+            expect(response.body.msg).toBe('Success')
+        })
+    })
+    
+})
+describe('GET /api/topics', () => {
+    test('get  /api/topics request reponds with topics object containing an array of topic objects ', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(response => {
+            const resBodyTopics = response.body.topics
+            expect(resBodyTopics).toBeInstanceOf(Array)
+            expect(resBodyTopics.length).not.toBe(0)
```

Comment:
We will also want to add an expect for what we want the length to be, to ensure all data has come through as expected. I would also suggest refactoring the for loop with a forEach (it may be easier to read)